<?xml version="1.0" encoding="UTF-8"?>
<launch>
     <arg name="hw_ns" default="RBRGS" />
     
     <!-- Fix Base TFs -->
     <!-- <node name="MoveitBase"
          pkg="robot_moveit_config" type="MoveItBase.py" output="screen"
     /> -->

     <!-- Run HardwareInterface -->
     <node name="MyRobot"
          pkg="robot_moveit_config" type="MyRobot" output="screen"
          ns="$(arg hw_ns)"
     />

     <!-- load controller configuration -->
     <rosparam command="load" file="$(find robot_moveit_config)/config/RBRGS_controllers.yaml" ns="$(arg hw_ns)" />
     <!-- spawn controller -->
     <node name="controllers_spawner_base" 
          pkg="controller_manager" type="spawner" output="screen" 
          args="base_controller"
          ns="$(arg hw_ns)"
     />
     <!-- <remap from="base_controller/follow_joint_trajectory" to="RBRGS/base_controller/follow_joint_trajectory"/> -->    
     <remap from="base_controller/follow_joint_trajectory" to="$(arg hw_ns)/base_controller/follow_joint_trajectory"/>
</launch>
